src/util/BinaryFuseFilter.cpp

```diff
@@ -0,0 +1,75 @@
+// Copyright 2024 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "util/BinaryFuseFilter.h"
+#include "util/siphash.h"
+#include <algorithm>
+#include <xdrpp/marshal.h>
+
+namespace stellar
+{
+
+template <typename T, typename U>
+BinaryFuseFilter<T, U>::BinaryFuseFilter(std::vector<uint64_t>& keyHashes,
+                                         binary_fuse_seed_t const& seed)
+    : mFilter(keyHashes.size()), mInputSeed(seed)
+{
+    // If too many hash collisions occur, population will fail. Retry with
+    // a different seed. This is unlikely to happen once, and is statically
+    // impossible to happen 10 times.
```

Comment:
"statically" impossible or _statistically_ impossible? In the paper it looks to me that a failure is in the 1-in-10 to 1-in-100 range -- so I'll buy it's statistically unlikely! But impossible seems strong.

Reply:
Yeah, statistically impossible. The library implementation itself will rotate the key 100 times then give up. Our wrapper around the library will rotate a different byte of the key then the library and try again 10 times, for a total seed rotation of 10 * 100. According to the original implementation, 100 seed rotations was already lower than the [chance of being hit with a cosmic ray](https://github.com/FastFilter/xor_singleheader/blob/3c0fd15b1e04281b2ada00cf82ddffc4b3292dee/include/binaryfusefilter.h#L308), so I think 10 * 100 can be considered statistically impossible.

Reply:
ok